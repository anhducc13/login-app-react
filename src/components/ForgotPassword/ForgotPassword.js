import React, { useState } from 'react';
import {
  Form,
  Input,
  Button,
  Col,
  Row,
} from 'antd';
import { Link } from 'react-router-dom';
import './ForgotPassword.scss';
import Title from 'antd/lib/typography/Title';
import { authServices } from 'services';
import openNotificationWithIcon from 'helpers/notification';
import { RULES_USERNAME, RULES_EMAIL } from 'constants/RuleValidators';


const ForgotPassword = (props) => {
  const bodyLayout = {
    xs: { span: 22, offset: 1 },
    sm: { span: 16, offset: 4 },
    md: { span: 12, offset: 6 },
    lg: { span: 10, offset: 7 },
    xl: { span: 8, offset: 8 },
    xxl: { span: 6, offset: 9 },
  }
  const [loading, setLoading] = useState(false);
  const { getFieldDecorator, validateFields, resetFields } = props.form;

  const handleSubmit = (e) => {
    e.preventDefault();
    validateFields((error, values) => {
      if (!error) {
        setLoading(true);
        const params = {
          "username": values.username,
          "email": values.email,
        }
        authServices.forgotPasswordUser(params)
          .then(() => {
            setLoading(false);
            openNotificationWithIcon('success', 'Success',
              `Please check email ${values.email} to receive new password`)
            props.history.push('/login');
          })
          .catch(err => {
            setLoading(false);
            if (err && err.response) {
              const { status } = err.response;
              if (status === 400) {
                resetFields();
                return;
              }
              if (status === 404) {
                props.history.push('/404');
                return;
              }
              if (status === 403) {
                openNotificationWithIcon('error', 'Error',
                  'Account has been block. Please try after')
                return;
              }
              if (status > 400 && status < 500) {
                props.history.push('/403');
                return;
              }
            }
            props.history.push('/500');
          })
      }
    })
  }



  return (
    <div className="ForgotPass">
      <Col {...bodyLayout}>
        <Form className="form-box">
          <Title
            level={2}
            style={{
              padding: "5px",
              textAlign: "center"
            }}
          >
            Forgot Password
          </Title>
          <Form.Item label="E-mail">
            {getFieldDecorator('email', {
              rules: RULES_EMAIL,
              validateFirst: true,
              validateTrigger: null,
            })(
              <Input size="large" />
            )}
          </Form.Item>
          <Form.Item label="Username">
            {getFieldDecorator('username', {
              rules: RULES_USERNAME,
              validateFirst: true,
              validateTrigger: null,
            })(
              <Input size="large" />
            )}
          </Form.Item>
          <Row>
            <Col style={{ textAlign: "center" }}>
              <Link to="/login">Back to login page?</Link>
            </Col>
          </Row>

          <Form.Item>
            <Button
              type="primary"
              htmlType="submit"
              block
              loading={loading}
              onClick={handleSubmit}
              size="large"
              style={{
                marginTop: 10,
              }}
            >
              Submit
            </Button>
          </Form.Item>
        </Form>
      </Col>
    </div>
  );
};

export default Form.create()(ForgotPassword);
