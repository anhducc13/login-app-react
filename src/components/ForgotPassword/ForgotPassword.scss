.ForgotPass {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  z-index: 1;
}

.form-box {
  margin-top: 60px;
  background-color: #ffffff;
  z-index: 10;
  padding: 10px 20px;
  border-radius: 10px;
  box-shadow: 0 1px 4px 0 rgba(0, 0, 0, 0.5);
  flex-direction: column;
}
