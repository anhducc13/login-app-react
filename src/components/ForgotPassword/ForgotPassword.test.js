import React from 'react';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import ForgotPassword from './ForgotPassword';

describe('Forgot Password Page Render', () => {
  it('should render correctly', () => {
    const component = shallow(<ForgotPassword />);
    expect(component).toMatchSnapshot();
  });
});

describe('Forgot Password form validate', () => {
  it('validate test case 1', () => {
    const component = mount(
      <MemoryRouter>
        <ForgotPassword />
      </MemoryRouter>
    );
    component.find('input[name="email"]').simulate('change', { target: { value: "" } });
    component.find('input[name="username"]').simulate('change', { target: { value: "anhducc14" } });
    component.find('button[type="submit"]').simulate('click');
    expect(component.find('div.ant-form-explain').first().text())
      .toBe("Please enter email")
    component.unmount()
  })

  it('validate test case 2', () => {
    const component = mount(
      <MemoryRouter>
        <ForgotPassword />
      </MemoryRouter>
    );
    component.find('input[name="email"]').simulate('change', { target: { value: "ab" } });
    component.find('input[name="username"]').simulate('change', { target: { value: "" } });
    component.find('button[type="submit"]').simulate('click');
    expect(component.find('div.ant-form-explain').length)
      .toBe(1)
    expect(component.find('div.ant-form-explain').first().text())
      .toBe("Please enter username")
    component.unmount()
  })

  it('validate test case 3', () => {
    const component = mount(
      <MemoryRouter>
        <ForgotPassword />
      </MemoryRouter>
    );
    component.find('input[name="email"]').simulate('change', { target: { value: "d" } });
    component.find('input[name="username"]').simulate('change', { target: { value: "a" } });
    component.find('button[type="submit"]').simulate('click');
    expect(component.find('div.ant-form-explain').length)
      .toBe(0)
    component.unmount()
  })
});