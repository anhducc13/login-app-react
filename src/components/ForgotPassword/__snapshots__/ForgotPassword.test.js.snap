// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forgot Password Page Render should render correctly 1`] = `
<div
  className="ForgotPass"
>
  <Col
    lg={
      Object {
        "offset": 7,
        "span": 10,
      }
    }
    md={
      Object {
        "offset": 6,
        "span": 12,
      }
    }
    sm={
      Object {
        "offset": 4,
        "span": 16,
      }
    }
    xl={
      Object {
        "offset": 8,
        "span": 8,
      }
    }
    xs={
      Object {
        "offset": 1,
        "span": 22,
      }
    }
    xxl={
      Object {
        "offset": 9,
        "span": 6,
      }
    }
  >
    <Form
      className="form-box"
      colon={true}
      hideRequiredMark={false}
      layout="horizontal"
      onSubmit={[Function]}
    >
      <Title
        level={2}
        style={
          Object {
            "padding": "5px",
            "textAlign": "center",
          }
        }
      >
        Forgot Password
      </Title>
      <FormItem
        hasFeedback={false}
        help=""
        label="E-mail"
        validateStatus=""
      >
        <Input
          disabled={false}
          name="email"
          onChange={[Function]}
          size="large"
          type="text"
          value=""
        />
      </FormItem>
      <FormItem
        hasFeedback={false}
        help=""
        label="Username"
        validateStatus=""
      >
        <Input
          disabled={false}
          name="username"
          onChange={[Function]}
          size="large"
          type="text"
          value=""
        />
      </FormItem>
      <Row
        gutter={0}
      >
        <Col
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <Link
            to="/login"
          >
            Back to login page?
          </Link>
        </Col>
      </Row>
      <FormItem
        hasFeedback={false}
      >
        <Button
          block={true}
          ghost={false}
          htmlType="submit"
          loading={false}
          onClick={[Function]}
          size="large"
          style={
            Object {
              "marginTop": 10,
            }
          }
          type="primary"
        >
          Submit
        </Button>
      </FormItem>
    </Form>
  </Col>
</div>
`;
